src/utils.rs

```diff
@@ -37,14 +37,48 @@ pub(crate) fn map_path_to_uri(uri_or_sha_prefix: &str) -> std::result::Result<St
         Ok(Url::from_file_path(path).unwrap().to_string())
     } else {
         let store = Store::default();
-        if let Some(policy) = store.get_policy_by_sha_prefix(uri_or_sha_prefix)? {
+        if let Some(policy) = store.get_policy_by_uri(uri_or_sha_prefix)? {
             Ok(policy.uri.clone())
         } else {
             Err(LookupError::PolicyMissing(uri_or_sha_prefix.to_string()))
         }
     }
 }
 
+pub(crate) fn verify_policy_version(uri: &str) -> std::result::Result<(), LookupError> {
+    let store = Store::default();
+    if store.get_policy_by_uri(uri)?.is_none() {
+        let policies = store.list()?;
+        let base_uri = if uri.starts_with("registry://") {
+            if let Some((base, tag_or_digest)) = uri.rsplit_once(':') {
+                if !tag_or_digest.contains('/') && !tag_or_digest.contains('@') {
+                    base
+                } else {
+                    uri
+                }
+            } else {
+                uri
+            }
```

Comment:
Parsing OCI references has many subtleties, I would use a [library](https://docs.rs/oci-spec/latest/oci_spec/distribution/struct.Reference.html) for that.

However I think we can drop this code, see my next comment